fix(antd): formItem tooltip 字符串被错误当作 props 处理的问题 close #4302

xiaosi-ss opened the pull request:
- 修复 isTooltipProps 函数逻辑，正确区分 ReactNode 和 Tooltip props
- 字符串、数字、React 元素等 ReactNode 类型现在被正确处理
- 只有包含 title、children 或 placement 等属性的对象才被当作 props
- 添加完整的单元测试覆盖各种 tooltip 类型场景

_Before_ submitting a pull request, please make sure the following is done...

- [x] Ensure the pull request title and commit message follow the [Commit Specific](https://formilyjs.org/guide/contribution#pr-specification) in **English**.
- [x] Fork the repo and create your branch from `master` or `formily_next`.
- [ ] If you've added code that should be tested, add tests!
- [ ] If you've changed APIs, update the documentation.
- [x] Ensure the test suite passes (`npm test`).
- [ ] Make sure your code lints (`npm run lint`) - we've done our best to make sure these rules match our internal linting guidelines.

**Please do not delete the above content**

---

## What have you changed?

![CleanShot 2025-07-08 at 17 32 02](https://github.com/user-attachments/assets/5b1955d3-9cfb-4b6d-aaa7-cc00eb0ff8c6)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/alibaba/formily?pullRequest=4314) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/alibaba/formily?pullRequest=4314) before we can accept your contribution.<br/><hr/>**xiaosi** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/alibaba/formily?pullRequest=4314) it.</sub>

xiaosi-ss committed:
fix(antd): formItem tooltip 字符串被错误当作 props 处理的问题 close #4302

- 修复 isTooltipProps 函数逻辑，正确区分 ReactNode 和 Tooltip props
- 字符串、数字、React 元素等 ReactNode 类型现在被正确处理
- 只有包含 title、children 或 placement 等属性的对象才被当作 props
- 添加完整的单元测试覆盖各种 tooltip 类型场景
